cmd/cayley/command/repl.go

```diff
@@ -101,9 +102,19 @@ func NewQueryCmd() *cobra.Command {
 		Aliases: []string{"qu"},
 		Short:   "Run a query in a specified database and print results.",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
-				return fmt.Errorf("expected query as an argument")
+			var querystr string
+			if len(args) == 0 {
+				bytes, err := ioutil.ReadAll(os.Stdin)
+				if err != nil {
+					panic(err)
+				}
+				querystr = string(bytes)
+			} else if len(args) == 1 {
+				querystr = args[0]
+			} else {
+				return fmt.Errorf("Query accepts only one argument, the query string or nothing for reading from stdin.")
 			}
+			fmt.Fprintf(os.Stderr, "Query to run is :%s\n", querystr)
```

Comment:
This should be `log.Printf`, since you use stderr now anyway. Also change a message to something simple, like `Query:` or `Running:`. Also it's a good idea to place line break after `:`.